src/eval.c

```diff
@@ -3605,6 +3618,282 @@ get_lit_string_tv(char_u **arg, typval_T *rettv, int evaluate)
     return OK;
 }
 
+    static int
+is_closing_quote(char_u *template, int is_literal_string, int *is_skipping_needed)
+{
+    char_u quote = is_literal_string ? '\'' : '"';
+
+    if (is_escaped_quote(is_literal_string, template))
+    {
+	*is_skipping_needed = TRUE;
+	return FALSE;
+    }
+
+    *is_skipping_needed = FALSE;
+    return *template == quote;
+}
+
+/*
+ * Allocate a variable for $"${var}" and $'${val}' constant.
+ * Return OK or FAIL.
+ */
+    static int
+get_template_string_tv(char_u **arg, typval_T *rettv, int evaluate)
+{
+    garray_T	plain;  // a string that is not a template string
+    char_u	*orig_arg = *arg;
+    int		is_literal_string = (*(*arg + 1) == '\'');
+    char_u	quote = is_literal_string ? '\'' : '"';
```

Comment:
How about this?
```C
    char_u	quote = *(*arg + 1);
    int		is_literal_string = (quote == '\'');
```
A bit simpler, I think.

Reply:
Good!
I'll fix it.